Assets/Scripts/ShadowPuzzle/PilarCheck.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor.Experimental.GraphView;
+using UnityEngine;
+
+public class PilarCheck : MonoBehaviour
+{
+    public bool pilarInPlace;
+    public GameObject pilarPlace;
+    public GameObject pilar;
+
+    void OnTriggerEnter(Collider other)
+    {
+        if (other.gameObject == pilar)
+        {
+            pilarInPlace = true;
+            FreezeMovement(other.gameObject);
+        }
+    }
+    public void FreezeMovement(GameObject obj)
+    {
+        {
+            Rigidbody rigidbody = obj.GetComponent<Rigidbody>();
+            if (rigidbody != null)
+            {
+                rigidbody.constraints = RigidbodyConstraints.FreezePositionX |
+                                         RigidbodyConstraints.FreezePositionY |
+                                         RigidbodyConstraints.FreezePositionZ;
+            }
+        }
```

Comment:
```suggestion
            Rigidbody rigidbody = pillar.GetComponent<Rigidbody>();
            if (rigidbody == null) return
            
                rigidbody.constraints = RigidbodyConstraints.FreezePositionX |
                                         RigidbodyConstraints.FreezePositionY |
                                         RigidbodyConstraints.FreezePositionZ;
            
        }
```
We hoeven geen gameobject mee te geven via the method we kunnen gewoon de gameobject gebruiken die in de class al opgeslages is bovenaan.

Je kan ook om de positie still te maken beter ook isKinematic gebruiken ipv manually alle posities te freezen.